prismarine-viewer/viewer/lib/entities.ts

```diff
@@ -547,10 +610,77 @@ export class Entities extends EventEmitter {
           nameTagScale: textDisplayMeta?.scale, nameTagTranslation: textDisplayMeta && (textDisplayMeta.translation || new THREE.Vector3(0, 0, 0)),
           nameTagRotationLeft: toQuaternion(textDisplayMeta?.left_rotation), nameTagRotationRight: toQuaternion(textDisplayMeta?.right_rotation) },
         this.entitiesOptions,
-        this.entities[entity.id].children.find(c => c.name === 'mesh')
+        mesh
       )
     }
 
+    const armorStandMeta = getSpecificEntityMetadata('armor_stand', entity)
+    if (armorStandMeta) {
```

Comment:
i think we have to split this huge update function into other functions like updateThing at some point

Reply:
Yeah, I have been thinking about how to better handle this too. Having a section for each entity would really bloat it, maybe it even needs to be in a separate file for each entity to properly support all of the functionality.

Reply:
Oh yes thats a good idea!